### DISCUSSION TITLE
[feature/ios] support self signed https URL

### DISCUSSION THREAD

**Discussion by lh99998888** [TheWidlarzGroup/react-native-video#4947] [Ideas]
### What problem are you trying to solve?

When I want to play a video asset stored in a self-signed certificate server via iOS, it shows an error: "The certificate for this server is invalid." I need a way to ignore certificate validation so that videos hosted on self-signed HTTPS servers can be played.

### Proposed solution

Add a way to ignore/bypass SSL certificate validation for self-signed HTTPS URLs on iOS, exposed through the component. When requesting data from a self-signed cert server with `NSURLSession`, certificate validation can be ignored using the following code:

```
- (void)URLSession:(NSURLSession *)session didReceiveChallenge:(NSURLAuthenticationChallenge *)challenge completionHandler:(void (^)(NSURLSessionAuthChallengeDisposition, NSURLCredential * _Nullable))completionHandler {
    NSURLCredential *credential = [NSURLCredential credentialForTrust:challenge.protectionSpace.serverTrust];
    completionHandler(NSURLSessionAuthChallengeUseCredential, credential);
}
```

How can I do this in this component?

### Platforms

iOS

### Additional context

Related to issue #2649.

**Comment by emreer94** [comment]
Hi, did you managed to resolve this problem? @lh99998888

[a comment by an automated account is not shown]

**Comment by TWG-AI** [comment]
Thanks for the idea — noted for the team to consider.

> I'm an AI assistant, so double-check any specifics I may have mentioned.
